akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -179,7 +175,21 @@ object ShardRegion {
   /**
    * Reply to `GetCurrentRegions`
    */
-  @SerialVersionUID(1L) final case class CurrentRegions(regions: Set[Address]) {
+  @SerialVersionUID(2L) final case class CurrentRegions(regionActorPaths: Set[ActorPath]) {
```

Comment:
I'm thinking about what API we want to provide. We don't want one single query to get all regions and all their entities, because that would result in huge messages (when millions of entities)). With similar reasoning I'm not sure about handing out the paths (refs) to the regions in this query, because then user would continue that by asking all regions for all entities and we would send a lot of data over the network.

Instead I think we should encourage local tasks. Let's say the user want to get an aggregated summary of number of entities on each node. Then it would be enough to hand out the Address of the regions here, and then they can have their own aggregation actor on each node that queries the local region for the entity information.

Perhaps not convenient but makes user's think before abusing.

wdyt?

Note that coordinator and region actors are for one entity type, so user have to know the entity types anyway.